File: app/src/main/java/com/anokmik/tripassistant/user/UserPresenter.java
package com.anokmik.tripassistant.user;

import com.anokmik.persistence.model.User;
import com.anokmik.persistence.repository.UserRepository;

public final class UserPresenter implements UserContract.Presenter {

    private final UserContract.View view;
    private final UserRepository userRepository;

    public UserPresenter(UserContract.View view) {
        this.view = view;
        this.userRepository = new UserRepository();
    }

    @Override
    public void validateUser() {
        User user = userRepository.getActive();
        if (user != null) {
            view.showProfile(user);
        } else {
            view.showLogin();
        }
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/user/UserPresenter.kt
package com.anokmik.tripassistant.user

import com.anokmik.persistence.repository.UserRepository

class UserPresenter(private val view: UserContract.View) : UserContract.Presenter {

    private val userRepository = UserRepository()

    override fun validateUser() {
        val user = userRepository.getActive()
        if (user != null) view.showProfile(user) else view.showLogin()
    }

}